### PR TITLE
Peter/refactor pre post pred

### PR DESCRIPTION
**Pull request by probicheaux**
# Description

Refactor core models breaking them into consistent preprocess, predict, postprocess interface

### CODE REVIEW

File: requirements/_requirements.txt [hunk 1]
```diff
@@ -3,6 +3,8 @@ cython<=3.0.0
 fastapi<=0.85.1
 numpy<=1.25.2
 opencv-python<=4.8.0.76
```
Comment (paulguerrie) [inline requirements/_requirements.txt:5]: Is this line redundant with the below headless install?

**Review by paulguerrie — APPROVED** [review]
Looks great! Much cleaner now!

File: inference/core/models/classification_base.py [hunk 2]
```diff
@@ -150,32 +150,57 @@ def infer(
             - The input image(s) will be preprocessed (normalized and reshaped) before inference.
             - This function uses an ONNX session to perform inference on the input image(s).
         """
-        t1 = perf_counter()
+        return super().infer(
```
[the remark this answers is not in the document]
Comment (probicheaux) [reply inference/core/models/classification_base.py:153]: yep, exactly

File: requirements/_requirements.txt (hunk 1 above)
Comment (probicheaux) [reply requirements/_requirements.txt:5]: I'm not sure, but I've seen some problems with differing version of opencv vs opencv headless so I'd just rather be sure

**Review by paulguerrie — APPROVED** [review]

**Pull request merged**